recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -148,11 +155,11 @@ add_task(withMockExperiments(withMockPreferences(async function (experiments, mo
     expired: false,
     preferenceName: "fake.preference",
     preferenceValue: "newvalue",
+    preferenceType: "string",
     previousPreferenceValue: "oldvalue",
     preferenceBranchType: "user",
   };
-  const experiment = {};
-  Object.keys(expectedExperiment).forEach(key => experiment[key] = experiments.test[key]);
+  const experiment = Object.assign({}, expectedExperiment);
```

Comment:
The original lines copied `experiments.test`, but only copied the keys from `experiment`. The intent was to only do a comparison of the values we care about, in case the experiment data ended up including more data later. The new line doesn't handle that, and should be changed back.

Although to be honest it's a confusing bit of code, and I should've left a comment mentioning that in the first place. My bad!